mbtiles/src/mbtiles.rs

```diff
@@ -39,6 +40,82 @@ pub struct Metadata {
     pub json: Option<JSONValue>,
 }
 
+#[derive(Clone, Debug, PartialEq, Serialize)]
+pub struct ZoomStats {
+    pub zoom: u8,
+    pub count: u64,
+    pub smallest: u64,
+    pub largest: u64,
+    pub average: f64,
+    pub bbox: Bounds,
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize)]
+pub struct Statistics {
+    pub file_path: String,
+    pub file_size: u64,
+    pub schema: MbtType,
+    pub page_size: u64,
+    pub zoom_stats_list: Vec<ZoomStats>,
+    pub count: u64,
+    pub smallest: Option<u64>,
+    pub largest: Option<u64>,
+    pub average: f64,
+}
+
+impl Display for Statistics {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        writeln!(f, "File: {}", self.file_path)?;
+
+        let file_size = SizeFormatterBinary::new(self.file_size);
+        writeln!(f, "FileSize: {file_size:.2}B")?;
+
+        writeln!(f, "Schema: {}", self.schema)?;
+
+        let page_size = SizeFormatterBinary::new(self.page_size);
+        writeln!(f, "Page size: {page_size:.2}B")?;
+
+        writeln!(
+            f,
+            "|{:^9}|{:^9}|{:^9}|{:^9}|{:^9}|{:^9}|",
+            "Zoom", "Count", "Smallest", "Largest", "Average", "BBox"
+        )?;
+
+        for l in &self.zoom_stats_list {
+            let smallest = SizeFormatterBinary::new(l.smallest);
+            let largest = SizeFormatterBinary::new(l.largest);
+            let average = SizeFormatterBinary::new(l.average as u64);
+
+            writeln!(
+                f,
+                "|{:>9}|{:>9}|{:>9}|{:>9}|{:>9}|{:>9}|",
+                l.zoom,
+                l.count,
+                format!("{smallest:.2}B"),
+                format!("{largest:.2}B"),
+                format!("{average:.2}B"),
+                l.bbox
+            )?;
+        }
+        if self.count != 0 {
+            let smallest = SizeFormatterBinary::new(self.smallest.expect("The smallest tile size of all zoom levels shouldn't be None when the tiles count of all zoom level is not 0"));
+            let largest = SizeFormatterBinary::new(self.largest.expect("The largest tile size of all zoom levels shouldn't be None when the tiles count of all zoom level is not 0"));
```

Comment:
no point for all this complex `expect` strings which will never happen. Might as well do a simpler variant:

```suggestion
        if let (Some(smallest), Some(largest)) = (self.smallest, self.largest) {
            let smallest = SizeFormatterBinary::new(smallest);
            let largest = SizeFormatterBinary::new(largest);
```

Reply:
Is `The largest tile size of all zoom levels shouldn't be None` Ok? Or just "Couldn't be None".

Reply:
you don't need to do any asserts (`.expected(...)`) - you can simply check if both values are not None, and if so, use them.